.github/workflows/build-and-test.yaml

```diff
@@ -0,0 +1,75 @@
+name: Build/Test
+
+on:
+  workflow_call:
+    secrets:
+      CHARMHUB_TOKEN:
+        required: true
+
+jobs:
+  lint:
+    name: Lint
+    runs-on: ubuntu-22.04
+    steps:
+      - name: Checkout
+        uses: actions/checkout@f43a0e5ff2bd294095638e18286ca9a3d1956744 # v3
+      - name: Install dependencies
+        run: python3 -m pip install tox
+      - name: Run linters
+        run: tox -e lint
+
+  unit-test:
+    name: Unit tests
+    runs-on: ubuntu-22.04
+    steps:
+      - name: Checkout
+        uses: actions/checkout@f43a0e5ff2bd294095638e18286ca9a3d1956744 # v3
+      - name: Install dependencies
+        run: python -m pip install tox
+      - name: Run tests
+        run: tox -e unit
+
+  lib-check:
+    name: Check libraries
+    runs-on: ubuntu-22.04
+    steps:
+      - name: Checkout
+        uses: actions/checkout@f43a0e5ff2bd294095638e18286ca9a3d1956744 # v3
+        with:
+          fetch-depth: 0
+      - name: Check libs
+        uses: canonical/charming-actions/check-libraries@631c2d944da2bd12430f1f3a954c8fffcf2385cd # 2.4.0
+        with:
+          credentials: "${{ secrets.CHARMHUB_TOKEN }}"
+          github-token: "${{ secrets.GITHUB_TOKEN }}"
+
+  integration-test:
+    name: Integration tests
+    runs-on: ubuntu-22.04
+    needs:
+      - lint
+      - unit-test
+      - lib-check
+    strategy:
+      fail-fast: false
+      max-parallel: 6
+      matrix:
+        agent-versions:
+          - "3.1.5"  # renovate: latest juju 3
```

Comment:
We should test with 2.9, 3.1, and latest feature release.

Reply:
@jat-canonical I guess you have been waiting to update this 3.2 and 2.9 until integration tests work.